From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: khaliforce <kforce@gmail.com>
Date: Mon, 26 Oct 2026 08:45:32 +0900
Subject: crypto: sun8i-ce - do not use the TRNG on D1

The D1 crypto engine has the first TRNG version (CE_ALG_TRNG).
Commit 4a07eab32f4b ("crypto: sun8i-ce - Add support for the TRNG")
keeps this version off on H3, H5, R40 and A64 because it fails rngtest.
The D1 output also fails all FIPS 140-2 tests, but hwrng credits it
as full entropy. Turn off the TRNG on D1, like on the other SoCs.

Tested on MangoPi MQ Pro (D1).

Fixes: 83f50f2948ba ("crypto: sun8i-ce - Add support for the D1 variant")
---
 drivers/crypto/allwinner/sun8i-ce/sun8i-ce-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/allwinner/sun8i-ce/sun8i-ce-core.c b/drivers/crypto/allwinner/sun8i-ce/sun8i-ce-core.c
index 21762bd408e5..8445213c8ee3 100644
--- a/drivers/crypto/allwinner/sun8i-ce/sun8i-ce-core.c
+++ b/drivers/crypto/allwinner/sun8i-ce/sun8i-ce-core.c
@@ -140,7 +140,7 @@ static const struct ce_variant ce_d1_variant = {
 		{ "trng", 0, 0 },
 		},
 	.esr = ESR_D1,
-	.trng = CE_ALG_TRNG,
+	.trng = CE_ID_NOTSUPP,
 };
 
 static const struct ce_variant ce_r40_variant = {
-- 
Armbian

